core/merginapi.cpp

```diff
@@ -3282,6 +3282,11 @@ void MerginApi::setServerType( const MerginServerType::ServerType &serverType )
 {
   if ( mServerType != serverType )
   {
+    if ( mApiRoot == defaultApiRoot() && mServerType == MerginServerType::OLD && serverType == MerginServerType::SAAS )
```

Comment:
I was playing around a little bit with it and in the end, we can safely remove the check for `defaultApiRoot` as we would not be able to test this feature. Thanks!

```suggestion
    if ( mServerType == MerginServerType::OLD && serverType == MerginServerType::SAAS )
```